src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1633,7 +1633,8 @@ private function generateMethodName($id)
             return $this->serviceIdToMethodNameMap[$id];
         }
 
-        $name = Container::camelize($id);
+        $i = strrpos($id, '\\');
+        $name = Container::camelize(false !== $i && isset($id[1 + $i]) ? substr($id, 1 + $i) : $id);
```

Comment:
Please add a test for a container using several class service ids with the same short class names, to avoid reintroducing the bug you had in the first version of the PR